TVM cleanup

vkovinicTT opened the pull request:
## Result

Compiler logic moved from `tvm` to `forge-fe`.

## Comprehensive overview

This is only the first cleaning pass where core compiler logic was transfered from tvm to the forge-fe. For comprehensive overview, TVM will be cleaned in a few stages.

**Coresponding PR in [tvm](https://github.com/tenstorrent/tt-tvm/pull/67).**

### Moved following files to `forge-fe` repo:
- `python/tvm/contrib/forge_compile.py`
- `python/tvm/contrib/forge_utils.py`
- `python/tvm/relay/op/contrib/forge/__init__.py`
- `python/tvm/relay/op/contrib/forge/forge.py`
- `python/tvm/relay/op/contrib/forge/relay_passes.py`
- `python/tvm/relay/op/contrib/forge/reportify.py`
- `python/tvm/relay/op/contrib/forge/utils.py`.

Coresponding structure:
```bash
forge/forge
│
├── tvm_calls
│   ├── forge_utils.py
│   ├── forge_compile.py
│   └── relay
│       ├── op
│       │   ├── forge_passes.py
│       │   ├── forge.py
│       │   ├── relay_passes.py
│       │   ├── reportify.py
│       │   └── utils.py
```

#### **Aditional notes:
These files were just moved to our repo so once uplift is done, we should probably refactor this code into more sensible modules.

nvukobratTT reviewed (approved):
Looks good. And I agree, we should restructure these once we complete with initial cleanup and uplift. 

Also, you have one minor issue with the updated location of `construct_pre_node_map`. Issues is rather trivial, so I'm still approving this PR so you can merge it once it's resolved.

Also, make sure to run nightly tests to confirm if there aren't some strange regressions coming out of this change.

vkovinicTT commented:
> Looks good. And I agree, we should restructure these once we complete with initial cleanup and uplift.
> 
> Also, you have one minor issue with the updated location of `construct_pre_node_map`. Issues is rather trivial, so I'm still approving this PR so you can merge it once it's resolved.
> 
> Also, make sure to run nightly tests to confirm if there aren't some strange regressions coming out of this change.

I've fixed the problem with `construct_pre_node_map`. After running the [nightly tests](https://github.com/tenstorrent/tt-forge-fe/actions/runs/13549368168), some have failed but not because of the TVM changes that I have introduced.
